### PR TITLE
Home Delivery Problem Reporting (case tweaks)

### PR DESCRIPTION
**Pull request by twrichards**
https://trello.com/c/1rXMTCPN/1007-hd-home-delivery-problem-reporting-case-assignment
image

The `Case Owner` changes are done in Salesforce - see https://github.com/guardian/salesforce/pull/184.

### CODE REVIEW

File: handlers/delivery-records-api/src/test/scala/com/gu/delivery_records_api/SFApiCompositeCreateDeliveryProblemTest.scala [hunk 1]
```diff
@@ -59,6 +59,7 @@ class SFApiCompositeCreateDeliveryProblemTest extends FlatSpec with Matchers {
          |        },
          |        "Origin" : "Self Service",
          |        "Status" : "New",
+         |        "Priority" : null,
          |        "Subject" : "[Self Service] Delivery Problem : No delivery (Guardian Weekly - A-S123456)",
```
Comment (mkuzdowicz) [inline handlers/delivery-records-api/src/test/scala/com/gu/delivery_records_api/SFApiCompositeCreateDeliveryProblemTest.scala]: `Priority` should that be None or null ?
why we set that to null fo we are using Option type ?

File: handlers/delivery-records-api/src/main/scala/com/gu/delivery_records_api/DeliveryRecordsService.scala [hunk 2]
```diff
@@ -57,7 +57,19 @@ case class CreateDeliveryProblem(
...
+  val isNotAutoCredit: Boolean = deliveryRecords.exists(_.invoiceDate.isEmpty)
+
+  val status: String = if (
+    isHomeDelivery ||
+    repeatDeliveryProblem.contains(true) ||
+    isNotAutoCredit
+  ) "New" else "Closed"
+
+  val priority: Option[String] = Some("High").filter(_ => isNotAutoCredit && isHomeDelivery)
+}
```
Comment (mkuzdowicz) [inline handlers/delivery-records-api/src/main/scala/com/gu/delivery_records_api/DeliveryRecordsService.scala]: so `High` and `None` are the only options ?

Comment (twrichards) [reply handlers/delivery-records-api/src/main/scala/com/gu/delivery_records_api/DeliveryRecordsService.scala]: see my other reply about this for more detail, but `High` is the only value we want to set it to explicitly, `None` is essentially 'not setting it' (as it was before)

File: handlers/delivery-records-api/src/test/scala/com/gu/delivery_records_api/SFApiCompositeCreateDeliveryProblemTest.scala (hunk 1 above)
Comment (mkuzdowicz) [reply handlers/delivery-records-api/src/test/scala/com/gu/delivery_records_api/SFApiCompositeCreateDeliveryProblemTest.scala]: ok

File: handlers/delivery-records-api/src/main/scala/com/gu/delivery_records_api/DeliveryRecordsService.scala (hunk 2 above)
Comment (mkuzdowicz) [reply handlers/delivery-records-api/src/main/scala/com/gu/delivery_records_api/DeliveryRecordsService.scala]: ok

**Pull request merged**